typescript/infra/scripts/deploy-igp.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  AbacusCore,
+  ChainMap,
+  ChainName,
+  CoreConfig,
+  CoreContracts,
+  MultiProvider,
+  objMap,
+  serializeContracts,
+} from '@abacus-network/sdk';
+
+import { AbacusCoreInfraDeployer } from '../src/core/deploy';
+import { readJSON, writeJSON } from '../src/utils/utils';
+
+import {
+  getCoreContractsSdkFilepath,
+  getCoreEnvironmentConfig,
+  getCoreVerificationDirectory,
+  getEnvironment,
+} from './utils';
+
+class IGPDeployer<
+  Chain extends ChainName,
+> extends AbacusCoreInfraDeployer<Chain> {
+  constructor(
+    multiProvider: MultiProvider<Chain>,
+    configMap: ChainMap<Chain, CoreConfig>,
+    public readonly existingContracts: ChainMap<
+      Chain,
+      CoreContracts<Chain, Chain>
+    >,
```

Comment:
still not sure I prefer this to the extending the "resumability" approach

Reply:
I agree that that is the right move eventually (i.e. idempotent deploys), but for now, lets keep it this way as the alternative seems like a lot of work?